tests/testthat/test_part3_no4am.R

```diff
@@ -0,0 +1,36 @@
+library(GGIR)
+context("g.part3")
+test_that("Part 3 runs fine on a short file not containing a night", {
+  skip_on_cran()
+  #=======================
+  # Create a test file that doesn't contain a night.
+  # More specifically, it should not contan a 4am, since it's a magic time used in g.sib.det
+  create_test_acc_csv(Nmin=1000, starts_at_midnight = FALSE)
+  fn = "123A_testaccfile.csv"
+  dn = "output_test"
+  if (file.exists(dn)) unlink(dn, recursive = TRUE)
+  # g.part3 modules report errors and warnings only by printing them out to the console,
+  # so these are not true errors and warnings that would get caught by the testing module.
+  # Instead, grab the console output and parse out any errors we should know about.
+  # 
+  # Note: make sure to call GGIR() with do.parallel = FALSE,
+  # otherwise not all console output will be captured.
+  out = capture.output(GGIR(mode = c(1:3), datadir = fn, outputdir = getwd(), studyname = "test", f0 = 1, f1 = 1,
+                                    do.report = c(), visualreport = FALSE, viewingwindow = 1,
+                                    overwrite = TRUE, do.parallel = FALSE, minimumFileSizeMB = 0), 
+                       type = c("output"))
+  
+  # There shouldn't be any errors or warnings reported, so if there are any, throw an error.
+  err_idx = grepl("Errors and warnings for", out)
+  for (ii in 1:length(err_idx)) {
+    if (err_idx[ii]) {
+      stop(out[ii:length(err_idx)])
+    }
+  }
+
```

Comment:
Can we replace this loop for something like this?
`if (any(err_idx)) stop(out[which(err_idx)])`
Or maybe:
`expect_false(any(err_idx))` to be consistent with the use of the testthat library

Reply:
Thank you, this does look much cleaner, I overuse loops all the time, ugh.

And thank you for drawing my attention to this logic. I used test_part3_1_night.R as a quick template for this test, because I wrote that test and knew how it worked, but I wasn't careful and didn't realize that error messaging in part3 got optimized since I wrote test_part3_1_night.R, here's the change: https://github.com/wadpac/GGIR/commit/535964ecb8eee0fd630bb45d97c1ab1be3ff95c0

Part 3 now throws errors in a normal way, instead of only printing them to the console. So there's no need to capture and parse console output anymore. I just removed the unnecessary code from both test_part3_1_night.R and test_part3_no4am.R